### DISCUSSION TITLE
Editor: The way Linear and Curve anchors can be added changed comparing to Stable

### DISCUSSION THREAD

**Discussion by Skubers** [ppy/osu#16805] [Q&A] [answered]
EDIT: Changed the title to correctly depict the discussion. More accurate info in the suggested answer!

As in the image provided. The anchor I picked is set to PerfectCurve, but it behaves like a linear. 

![osu_2022-02-06_18-32-01](https://user-images.githubusercontent.com/82095491/152693385-6d0bf8ab-f5cd-42eb-9cff-bec5084a22e9.jpg)

**Comment by peppy** [comment, marked as answer]
You are one point too short. A curve can't work with only two points.

**Reply by Skubers** [reply]
oh, that's entirely different than on stable, kinda unintuitive if you ask me

**Reply by peppy** [reply]
Stable also requires three points to make a curve.

Two points is a line...

Could you please provide a video showing what you mean if this is unintuitive to you?

**Reply by Skubers** [reply]
So in stable you determine if you want a sharp angle or a slope by using respectively red anchor or white anchor -in the place of a curve-

![screenshot128](https://user-images.githubusercontent.com/82095491/152782283-1e2c1918-edf1-4c7d-a373-976b9295efe3.jpg)

In lazer however, if you do that, nothing changes.

![osu_2022-02-07_12-43-01](https://user-images.githubusercontent.com/82095491/152782337-70ece997-1de1-4b49-897e-9d8de9c37752.jpg)

After some testing, I've noticed that you need to change the sliderhead's curve type to change the next curve which is not intuitive at all imo.

![osu_2022-02-07_12-46-32](https://user-images.githubusercontent.com/82095491/152782572-3511ffd1-eabd-403b-b341-e02f3dd174ae.jpg)

Additionally in the screenshot above, you can see that on the third slider, I've set PerfectCurve on both head and body anchors and for some reason they make a sharp angle. Is that intended?

**Reply by peppy** [reply]
Putting a "perfect curve" anchor in the middle of a curve makes no sense. It should be at the start of the curve, not in the middle. The behaviour hasn't changed from stable, but new features are added (which you are mistakenly placing in the wrong place).

**Reply by Skubers** [reply]
I think we don't understand each other well here, but I can't really explain it better than I've done it here. Maybe another mapper might jump on that discussion and get their opinion

**Reply by peppy** [reply]
I think you are misunderstanding something. Slider placement hasn't changed between stable and lazer. The difference is now you can have multiple curve types in a slider.

If you want stable behaviour, do not place any curve types manually. Use what you're used to and you'll get the old behaviour.

**Reply by Skubers** [reply]
I've done some videos of slider creation flow:
This is how I (and probably most people) do sliders on stable. We put a straight slider and add anchors in between to make either curve or an angle

https://user-images.githubusercontent.com/82095491/152817479-3a471c06-cd6b-4d6e-ac7e-82ba5ef93547.mp4

Now I'm doing the same thing on lazer and it only makes sharp curves. So the flow changed here

https://user-images.githubusercontent.com/82095491/152817649-f81c7126-c8ec-4177-9fb5-f2393a254997.mp4

**Reply by peppy** [reply]
That's not how slider placement was intended to be done.

You have two options:

- Place more than two points before ending placement (it will create a curve)
- Change the red point to a curve type, instead of leaving it linear.
